shared/src/ledger/governance/mod.rs

```diff
@@ -0,0 +1,493 @@
+                (KeyType::AUTHOR, Some(proposal_id)) => {
+                    let author_key = gov_storage::get_author_key(proposal_id);
+                    let author =
+                        read(&self.ctx, &author_key, ReadType::POST).ok();
+                    let has_pre_author = self.ctx.has_key_pre(&author_key).ok();
+                    match (has_pre_author, author) {
+                        (Some(has_pre_author), Some(author)) => {
+                            !has_pre_author && verifiers.contains(&author)
+                        }
+                        _ => false,
+                    }
+                }
+                (KeyType::COUNTER, _) => {
+                    let counter_key = gov_storage::get_counter_key();
+                    let pre_counter: Option<u64> =
+                        read(&self.ctx, &counter_key, ReadType::PRE).ok();
+                    let post_counter: Option<u64> =
+                        read(&self.ctx, &counter_key, ReadType::POST).ok();
+                    match (pre_counter, post_counter) {
+                        (Some(pre_counter), Some(post_counter)) => {
+                            pre_counter + 1 == post_counter
+                        }
+                        _ => false,
+                    }
+                }
+                (KeyType::BALANCE, _) => {
+                    let balance_key =
+                        token_storage::balance_key(&m1t(), &ADDRESS);
+                    let min_funds_parameter_key =
+                        gov_storage::get_min_proposal_fund_key();
+                    let min_funds_parameter: Option<Amount> = read(
+                        &self.ctx,
+                        &min_funds_parameter_key,
+                        ReadType::PRE,
+                    )
+                    .ok();
+                    let pre_balance: Option<Amount> =
+                        read(&self.ctx, &balance_key, ReadType::PRE).ok();
+                    let post_balance: Option<Amount> =
+                        read(&self.ctx, &balance_key, ReadType::POST).ok();
+                    match (min_funds_parameter, pre_balance, post_balance) {
+                        (
+                            Some(min_funds_parameter),
+                            Some(pre_balance),
+                            Some(post_balance),
+                        ) => {
+                            post_balance > pre_balance
+                                && post_balance - pre_balance
+                                    >= min_funds_parameter
+                        }
+                        (
+                            Some(min_funds_parameter),
+                            None,
+                            Some(post_balance),
+                        ) => post_balance >= min_funds_parameter,
+                        _ => false,
+                    }
+                }
+                (KeyType::PARAMETER, _) => false,
+                (KeyType::UNKNOWN, _) => false,
```

Comment:
I think in order to avoid having to change this again for https://github.com/anoma/anoma/issues/955 (and in general in other VPs), we could have two cases here:
- an unknown key type that is within this governance address space (the first segment == this address), that is rejected
- and another case for unknown keys from other accounts, which this VP shouldn't care about and so it should be permissive returning `true`